File: Gauntlet of the Hero - GOTH/Assets/_Scripts/Models/Backend/RunResolveEncounterResponseDto.cs
using Newtonsoft.Json;

namespace GauntletOfTheHero.Backend.Models
{
    [System.Serializable]
    public class RunResolveEncounterResponseDto
    {
        [JsonProperty("run")]
        public RunConfigDto run;

        [JsonProperty("learnedMove")]
        public MoveDto learnedMove;

        [JsonProperty("xpGained")]
        public int xpGained;

        [JsonProperty("levelsGained")]
        public int levelsGained;
    }
}

File: nordeus_challenge_backend_spring_boot/src/main/java/com/mmileticc/nordeuschallenge/model/dto/RunResolveEncounterResponseDTO.java
package com.mmileticc.nordeuschallenge.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RunResolveEncounterResponseDTO {

    private RunConfigDTO run;
    private MoveDTO learnedMove;
    private int xpGained;
    private int levelsGained;
}